amundsen_application/static/js/components/common/BadgeList/index.tsx

```diff
@@ -4,36 +4,122 @@
+import { BadgeStyle, BadgeStyleConfig } from 'config/config-types';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
 
-export interface BadgeListProps {
+// import Flag, { FlagProps, convertText, CaseType } from 'components/common/Flag';
+import { ResourceType } from 'interfaces';
+import { updateSearchState } from 'ducks/search/reducer';
+import { UpdateSearchStateRequest } from 'ducks/search/types';
+import { logClick } from 'ducks/utilMethods';
+import './styles.scss';
+
+const COLUMN_BADGE_CATEGORY = 'column';
+
+export interface ListProps {
   badges: Badge[];
 }
 
-const BadgeList: React.FC<BadgeListProps> = ({ badges }: BadgeListProps) => {
+export interface DispatchFromProps {
+  searchBadge: (badgeText: string) => UpdateSearchStateRequest;
+}
+
+export interface ActionableBadgeProps {
+  style: BadgeStyle,
+  displayName: string,
+  action: any;
+}
+
+export type BadgeListProps = ListProps & DispatchFromProps;
+
+
+const StaticBadge: React.FC<BadgeStyleConfig> = ({ style, displayName }: BadgeStyleConfig) => {
   return (
-    <span className="badge-list">
-      {badges.map((badge, index) => {
-        let badgeConfig;
-        // search badges with just name
-        if (badge.tag_name) {
-          badgeConfig = getBadgeConfig(badge.tag_name);
-        }
-        // metadata badges with name and category
-        else if (badge.badge_name) {
-          badgeConfig = getBadgeConfig(badge.badge_name);
+    <span className={`static-badge flag label label-${style}`}>
+      <div className={`badge-overlay-${style}`}>
+        {displayName}
+      </div>
+    </span>);
+};
+
+const ActionableBadge: React.FC<ActionableBadgeProps> = ({ style, displayName, action }: ActionableBadgeProps) => {
+  return (
+    <span className="actionable-badge" onClick={action}>
+      <StaticBadge
+        style={style}
+        displayName={displayName}
+      />
+    </span >
+  );
+};
+
+export class BadgeList extends React.Component<BadgeListProps> {
+  render() {
+    return (
+      <span className="badge-list">
+        {this.props.badges.map((badge, index) => {
+          let badgeConfig;
+          let originalBadgeName;
+          // search badges with just name
+          if (badge.tag_name) {
+            originalBadgeName = badge.tag_name;
+            badgeConfig = getBadgeConfig(badge.tag_name);
+          }
+          // metadata badges with name and category
+          else if (badge.badge_name) {
+            originalBadgeName = badge.badge_name;
+            badgeConfig = getBadgeConfig(badge.badge_name);
+            if (badge.category === COLUMN_BADGE_CATEGORY) {
+              return (<StaticBadge
+                style={badgeConfig.style}
+                displayName={badgeConfig.displayName}
+                key={`badge-${index}`}
+              />);
+            }
+          }
+          if (badge.category !== COLUMN_BADGE_CATEGORY) {
+            let onClick = (e) => {
```

Comment:
Let's rename this to handleClick, following our new naming convention.

Also, this handler should be out of the map, so we don't create a new handler for each badge of the list